### PR TITLE
[Merged by Bors] - feat: cyclic group lemmas

### PR DESCRIPTION
**Pull request by fpvandoorn**
* Add a lemma stating that non-identity element of a finite group of prime order generates the group (stated in 4 ways)
* Use this to golf `isCyclic_of_prime_card`
* Add a version of `isUnit_iff_exists_inv` for non-commutative monoids.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by fpvandoorn** [comment]
This will conflict with #10832. If one is merged, the other PR should merge master.

File: Mathlib/Algebra/Group/Units.lean
Comment (adomani) [inline Mathlib/Algebra/Group/Units.lean:646]: Is this the same lemma as the one in #10719 (not merged, yet)?

File: Mathlib/GroupTheory/SpecificGroups/Cyclic.lean
Comment (adomani) [inline Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:149]:
```suggestion
theorem mem_powers_of_prime_card {G : Type*} [Group G] {_ : Fintype G} {p : ℕ} [hp : Fact p.Prime]
```
or is there a reason for using the explicit universe?

**Review by adomani — APPROVED** [review]
I imagine that the `to_additive` names explicitly provided here will become superfluous once #10831 gets merged, right?

If so, then I think that it might be better to wait for the other PR and merge this one afterward, with the enhanced `to_additive` translations.

File: Mathlib/Algebra/Group/Units.lean
Comment (fpvandoorn) [reply Mathlib/Algebra/Group/Units.lean:646]: Not, exactly (one existential vs two), but I might be happy with that one as well.

**Comment by fpvandoorn** [comment]
Yes, there is a conflict. I don't care which PR gets merged first.

File: Mathlib/Algebra/Group/Units.lean
Comment (adomani) [reply Mathlib/Algebra/Group/Units.lean:646]: Ah, of course!  I am happy with having both lemmas, although ideally they should end up next to one another and, even better, have doc-strings mentioning each other.  This may simply be the job of a PR following the merging of the two existing PRs.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/GroupTheory/SpecificGroups/Cyclic.lean
Comment (riccardobrasca) [inline Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:144]: Why are you using `{_ : Fintype G}` instead of `[Fintype G]`?

Comment (fpvandoorn) [reply Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:144]: Then it can sometimes be figured out from the type of `h`, even if this argument is not an instance. I don't think it's very important.

**Pull request closed without merging**
